### DISCUSSION TITLE
API Structure

### DISCUSSION THREAD

**Discussion by guidiego** [vercel/next.js#18923] [Help] [answered]
Hi I am trying to purpose an issue/solution, but I am with some problems in understanding how `/api` works, could someone help me through the code. I found some code about load pages, but nothing about load API :) Unsure if it shares the same code with different entry points

**Comment by jamesmosier** [comment, marked as answer]
Hi @guidiego. I'm not sure exactly what you are looking for, but if you are interested in learning more about how Next handles API routes you can see this function which has some API logic:

https://github.com/vercel/next.js/blob/9a13dd3466ba617147a03be1d5e5b2967c10d0f0/packages/next/next-server/server/next-server.ts#L891-L896

as well as 

https://github.com/vercel/next.js/blob/91b5390617a1dac36a4ea0cebcb2a21a06630f0a/packages/next/next-server/server/api-utils.ts#L23-L31

**Reply by guidiego** [reply]
Hey @jamesmosier amazing! A question: https://github.com/vercel/next.js/blob/9a13dd3466ba617147a03be1d5e5b2967c10d0f0/packages/next/next-server/server/next-server.ts#L901

If I want to make one more page like `/foo` to handle another kind of things, what you think is the better approach?

**Reply by guidiego** [reply]
Unsure about how dynamicRoutes work as well

**Reply by jamesmosier** [reply]
I'm not certain, but I believe you'll want to start here if you want to handle a route such as `/foo`:

https://github.com/vercel/next.js/blob/9a13dd3466ba617147a03be1d5e5b2967c10d0f0/packages/next/next-server/server/next-server.ts#L817
